resources/js/Components/Form/DmEntryCreateForm.tsx

```diff
@@ -0,0 +1,479 @@
+import {Link, useForm, usePage} from '@inertiajs/inertia-react'
+import AdapterDateFns from '@mui/lab/AdapterDateFns'
+import DatePicker from '@mui/lab/DatePicker'
+import LocalizationProvider from '@mui/lab/LocalizationProvider'
+import {
+    Box,
+    Button,
+    Grid,
+    Step,
+    StepLabel,
+    Stepper,
+    TextField,
+    Typography,
+} from '@mui/material'
+import Alert from '@mui/material/Alert'
+import Chip from '@mui/material/Chip'
+import FormControl from '@mui/material/FormControl'
+import MenuItem from '@mui/material/MenuItem'
+import {ErrorText} from 'Components'
+import React, {useState} from 'react'
+import styled from 'styled-components'
+import {EntriesData} from 'Types/entries-data'
+import {UsePageType} from 'Types/global'
+import {ItemData} from 'Types/item-data'
+import route from 'ziggy-js'
+
+type DmEntryCreateFormPropType = {
+    type: 'Edit' | 'Create'
+    onCloseDrawer?: () => void
+    editData?: EntriesData
+    editId?: number
+}
+type ItemDataType = {
+    name: string
+    description: string | null
+    rarity: string
+    tier: number
+}
+
+type DmEntryFormDataType = {
+    adventure_id: number
+    length: number
+    location: string
+    date_played: string | null
+    notes: string
+    items: ItemData[]
+    type: string
+    user_id: number | null | undefined
+}
+
+const StyledBox = styled(Box)`
+    padding: 32px 0px 16px 0px;
+`
+
+const FormBox = styled(Box)`
+    flexgrow: 1;
+    margin-top: 16px;
+`
+
+const StyledGrid = styled(Grid)`
+    align-items: center;
+    justify-content: center;
+    margin-bottom: 16px;
+`
+const StyledDateGrid = styled(Grid)`
+    margin-top: 32px;
+    margin-bottom: 23px;
+`
+
+const StyledFooter = styled(Grid)`
+    min-width: 25vw;
+`
+const StyledItemsFooter = styled(Grid)`
+    margin-top: 16px;
+    min-width: 25vw;
+`
+
+const DmEntryCreateForm = ({
+    type,
+    onCloseDrawer = () => {},
+    editData,
+    editId = 0,
+}: DmEntryCreateFormPropType) => {
+    const {auth, adventures} = usePage<UsePageType>().props
```

Comment:
don't use `usePage`, pass down from parent component please,
`usePage` is only to use for global component that exists in every single page like `Authentication`